### PR TITLE
Rootlesskit Network Design

### PR DESCRIPTION
**Pull request by fahedouch**
[rendered doc](https://github.com/containerd/nerdctl/blob/41b83493e1e41b8a76db25f2dbfb5046edf685ee/docs/rootless.md)

### CODE REVIEW

File: docs/assets/images/rootlessKit-network-design.png
Comment (AkihiroSuda) [inline docs/assets/images/rootlessKit-network-design.png:1]: `docs/assets/images/rootlessKit-network-design.png` -> `docs/images/rootlesskit-network-design.png`

Comment (AkihiroSuda) [reply docs/assets/images/rootlessKit-network-design.png:1]: detach-netns should be explained in this figure?

File: docs/rootless.md [hunk 1]
```diff
@@ -160,6 +160,18 @@ systemctl --user daemon-reload
 systemctl --user restart containerd
 ```
 
+### Rootlesskit Network Design
```
Comment (AkihiroSuda) [inline docs/rootless.md:163]: This should be the last section

File: docs/rootless.md [hunk 2]
```diff
@@ -160,6 +160,18 @@ systemctl --user daemon-reload
...
+
+* `Nerdctl` supports rootlesskit `detach-netns` mode to ensure controlling network access for the container in detach mode. If using the non `detach-netns` mode, all steps mentioned below would be executed within the RootlessKit `Child` network namespace directly.
```
Comment (AkihiroSuda) [inline docs/rootless.md:165]: - s/Nerdctl/nerdctl/
- detach-netns mode needs explanation
-

File: docs/rootless.md [hunk 3]
```diff
@@ -160,6 +160,18 @@ systemctl --user daemon-reload
...
+
+![rootlessKit-network-design.png](assets/images/rootlessKit-network-design.png)
+
+- Rootlesskit Parent NetNS and Child NetNS are already configured by the startup script [containerd-rootless.sh](https://github.com/fahedouch/nerdctl/blob/3e5b0853067d38b2ab1a456bdc31fd08d0462aae/extras/rootless/containerd-rootless.sh)
+- Rootlesskit Parent NetNS is the host network namespace
+- step3: `runc` creates container with dedicated namespaces (e.g network ns) in the Parent netns
+- step4: `runc` nsenter Rootlesskit Child NetNS before triggering nerdctl ocihook
+- step6: CNI configures container network namespace: create network interfaces `eth0` -> `veth0` -> `nerdctl0`
```
Comment (AkihiroSuda) [inline docs/rootless.md:173]: Lacks step 1, 2, and 5

File: docs/rootless.md [hunk 4]
```diff
@@ -160,6 +160,18 @@ systemctl --user daemon-reload
 systemctl --user restart containerd
 ```
 
+### Rootlesskit Network Design
+
+* `Nerdctl` supports rootlesskit `detach-netns` mode to ensure controlling network access for the container in detach mode. If using the non `detach-netns` mode, all steps mentioned below would be executed within the RootlessKit `Child` network namespace directly.
+
+![rootlessKit-network-design.png](assets/images/rootlessKit-network-design.png)
+
+- Rootlesskit Parent NetNS and Child NetNS are already configured by the startup script [containerd-rootless.sh](https://github.com/fahedouch/nerdctl/blob/3e5b0853067d38b2ab1a456bdc31fd08d0462aae/extras/rootless/containerd-rootless.sh)
```
Comment (AkihiroSuda) [inline docs/rootless.md:169]: No need to use your fork

File: docs/assets/images/rootlessKit-network-design.png
Comment (fahedouch) [reply docs/assets/images/rootlessKit-network-design.png:1]: fixed

File: docs/rootless.md (hunk 1 above)
Comment (fahedouch) [reply docs/rootless.md]: fixed

File: docs/rootless.md (hunk 2 above)
Comment (fahedouch) [reply docs/rootless.md]: fixed

File: docs/rootless.md (hunk 3 above)
Comment (fahedouch) [reply docs/rootless.md]: it was trivial to me but ok if you want!

File: docs/rootless.md (hunk 4 above)
Comment (fahedouch) [reply docs/rootless.md]: fixed

File: docs/rootless.md [hunk 5]
```diff
@@ -164,3 +164,22 @@ systemctl --user restart containerd
...
+- For creating and configuring the container's network namespace, the child command switches temporarily to the relevant namespace located in `$ROOTLESSKIT_STATE_DIR/netns`. This ensures necessary network setup while maintaining isolation in the host namespace.
+
+![rootlessKit-network-design.png](images/rootlessKit-network-design.png)
+
+- Rootlesskit Parent NetNS and Child NetNS are already configured by the startup script [containerd-rootless.sh](https://github.com/containerd/nerdctl/blob/main/extras/rootless/containerd-rootless.sh)
+- Rootlesskit Parent NetNS is the host network namespace
+- step1: `nerdctl` calls `containerd` in the host network namespace.
+- step2: `containerd` calls `runc` in the host network namespace.
+- step3: `runc` creates container with dedicated namespaces (e.g network ns) in the Parent netns.
+- step4: `runc` nsenter Rootlesskit Child NetNS before triggering nerdctl ocihook.
```
Comment (AkihiroSuda) [inline docs/rootless.md:183]: nsenter is called by nerdctl, not by runc

**Pull request merged**